src/effects/builtin/noisegen/noiseviewmodel.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Audacity: A Digital Audio Editor
+ */
+#pragma once
+
+#include "../common/generatoreffectmodel.h"
+#include "../common/params.h"
+#include "noisegenerator.h"
+
+namespace au::effects {
+class NoiseGenerator;
+
+class NoiseViewModel : public GeneratorEffectModel
+{
+    Q_OBJECT
+
+    Q_PROPERTY(double amplitude READ amplitude WRITE setAmplitude NOTIFY amplitudeChanged)
+    Q_PROPERTY(NoiseSettings::Type type READ type WRITE setType NOTIFY typeChanged)
```

Comment:
hmm, does this work?
to use a type in Qml, it needs to be registered in the meta system.

Reply:
I believe that you need to register an enum if you want to expose literals. In this case it just decays to int. What should i do, register an enum or replace property type to int?

Reply:
I think in this case it is easier to use int.
no need to use enums from the domain model in view (or we need to make two enums and convert between them), so as not to relation the domain model with the implementation of view.